OrbitGl/TimeGraph.cpp

```diff
@@ -610,44 +668,59 @@ void TimeGraph::DrawOverlay(GlCanvas* canvas, bool picking) {
 
   double inv_time_window = 1.0 / GetTimeWindowUs();
 
-  for (const auto& current_textbox : overlay_current_textboxes_) {
-    const Timer& timer = current_textbox.second->GetTimer();
+  // Draw lines for iterators.
+  for (const auto& box : boxes) {
+    const Timer& timer = box.second->GetTimer();
+
     double start_us = GetUsFromTick(timer.m_Start);
     double normalized_start = start_us * inv_time_window;
     float world_timer_x =
         static_cast<float>(world_start_x + normalized_start * world_width);
 
     Vec2 pos(world_timer_x, world_start_y);
+    x_coords.push_back(pos[0]);
 
-    min_x = std::min(min_x, world_timer_x);
-    min_tick = std::min(min_tick, timer.m_Start);
-    max_x = std::max(max_x, world_timer_x);
-    max_tick = std::max(max_tick, timer.m_Start);
+    canvas->GetBatcher()->AddVerticalLine(
+        pos, -world_height, GlCanvas::Z_VALUE_OVERLAY,
+        GetThreadColor(timer.m_TID), PickingID::LINE, nullptr);
+  }
+
+  // Draw boxes with timings between iterators.
+  for (size_t k = 1; k < boxes.size(); ++k) {
+    Vec2 pos(x_coords[k - 1], world_start_y - world_height);
+    float size_x = x_coords[k] - pos[0];
+    Vec2 size(size_x, world_height);
+    Color color = GetIteratorBoxColor(k - 1);
+
+    auto& labels =
+        GetLabelBetweenIterators(boxes[k - 1].second, boxes[k].second);
 
-    float z = GlCanvas::Z_VALUE_OVERLAY;
-    Color color = GetThreadColor(timer.m_TID);
+    float text_y_offset = world_height / 2.f /
+                          static_cast<float>(overlay_current_textboxes_.size());
+    float text_y =
+        pos[1] + (world_height / 2.f) - static_cast<float>(k) * text_y_offset;
```

Comment:
I'm having issues here. Maybe is my linux workstation. I think that we have to add the margin to this world_height. It isn't divided in equal parts. In my linux view I had 30 of margin. So the real world_height (from WorldTopLeftY to the real bottom is 30 less).

Reply:
I'm pretty sure that this downMargin of 30 is exactly 2 * m_SliderWidth.

Reply:
I had a look and the definitions of the margins, etc. are not 100% clear to me. 

What is clear: The bottom shows a slider (TimeGraphLayout::GetSliderWidth) and there is also the timer bar with timing values. For the latter I added a new TimerGraphLayout value that specifies the height of the timer bar (previously GetSliderWidth() was used for that). So far, I think this is ok. (This also matches what you said about the down margin being 2 * m_SliderWidth.)

But just using these two values is not quite enough, the text at the bottom iterator still overlaps a little with the timer bar. 

When I add TimeGraphLayout::GetVerticalMargin(), it seems to be better, but I don't fully understand the meaning of this value. In one place it is used for the width of the right margin (see CaptureWindow::DrawScreenSpace).

Reply:
I'm also wondering if we should remove a margin at the top and possibly the scheduler track offset. WDYT?

Reply:
Main issue should be fixed now. 

@karupayun Do you have an opinion regarding how to handle to top margin, if at all?

Reply:
I think that m_Layout.GetSchedulerTrackOffset() is the topMargin.

Reply:
I prefer to have the total time (n > 2) at the 1/3 line (instead of 1/2) and divided the rest (2/3) for the iterators. But, having it at the 1/2 line is ok if you prefer that.
And, counting both top_margin and down_margin. top_margin is not too interesting now, but if you have to zoom could be.

Reply:
Yes, agree. Should I take it into account in the computation? That would push the "center" of where I draw "Total" down a bit. It doesn't really matter and the window can't be resized to be small enough that you would only see the scheduler track. And I'm not even sure how I want things to behave then.

Reply:
I'll merge for now, but let's chat if you think this still requires a change.